component/registry.go

```diff
@@ -28,9 +29,40 @@ var (
 	parsedNames = map[string]parsedName{}
 )
 
+// ModuleSystem is a mechanism responsible for allowing components to create other components, or
+// flow systems.
+type ModuleSystem interface {
+	// NewModuleDelegate creates a new, un-started ModuleDelegate.
+	NewModuleDelegate(id string) ModuleDelegate
+}
+
+// ModuleDelegate is a controller for running components within a Module.
+type ModuleDelegate interface {
```

Comment:
Not in love with the names here. What about `ModuleController` and `Module`? My concern with introducing "system" as a term is that people might get confused about if there's a semantic difference between an "X system" and an "X controller," even though the terms are used interchangeably. 

I don't think ModuleDelegate needs to be called a "delegate;" from the perspective of a user it's no different from a module, where they load the module source, run the module, and retrieve an HTTP handler representing the module's API. 

I'm more opinionated that "ModuleDelegate" should be "Module," but I can be convinced that ModuleSystem is good enough for now if there's no opportunity to get confused by both system and controller being used in different places to refer to the same thing.

A third option is to make `NewModule` a function in Options rather than passing an interface, though that would prevent us from ever adding more methods.

Reply:
I am not a fan of any of the names TBH, I can live with `controller` and `module`. I just hate how we have something like 11 files with controller in the name/type.